### DISCUSSION TITLE
Ajout de passerelle pour les serveurs logiques

### DISCUSSION THREAD

**Discussion by ghost** [dbarzin/mercator#112] [Ideas]
Bonjour.
 
Avant toute chose je voulais vous remercier et vous féliciter pour votre travail.
La cartographie de notre SI selon les normes de l'ANSSI est un défi fort complexe que nous souhaitions relever depuis longtemps.
Je test depuis quelque semaine Mercator et je pense personnellement que c'est d'or et déjà un excellent produit.
Je me demandais si vous aviez prévu d'ajouter la déclaration d'une passerelle pour les serveurs logiques.
Cela permettrait une meilleure hiérarchisation des éléments sur la vue logique.
Un bouton qui permet de dupliquer une entrée serait également une fonctionnalité fort appréciable de notre point de vue.

Qu'en pensez-vous ?

Je vous souhaite une excellente journée.
Et encore bravo et merci pour votre travail.

**Comment by dbarzin** [comment]
Bonjour,

Qu'entendez vous pas "la déclaration d'une passerelle pour les serveurs logiques".
C'est un autre serveur logique, un nouvel objet, une adresse IP ?
Pour le moment vous pouvez ajouter cette information dans le champ texte "configuration" d'un serveur logique.

Pour la fonction "clone", comme cela a déjà été demandé plusieurs fois, cette fonction a été ajoutée à la feuille de route.
Le bouton sera présent sur l'écran d'affichage d'un objet.

Merci,
Didier

**Comment by ghost** [comment]
Dans les passerelles nous pouvons déclarer des interfaces. Le champs "IP publique et privée"
Typiquement sur une infra trés basique on va retrouver les interfaces LAN et WAN.

Si nous pouvions déclarer une adresse IP dans le formulaire des serveurs logiques cela permettrait, si je ne dis pas de bêtises, de rajouter un lien hiérarchique entre les serveurs et leurs passerelles. 

Ainsi dans les représentations graphiques nous verrions apparaître un niveau supplémentaire.
Les serveurs et les sous-réseaux serait reliés derrière leurs firewalls respectifs.

Si je ne suis pas clair où que vous souhaitez des précisions n'hésitez pas.
Si vous pensez que c'est judicieux je peux également vous contacter de vive voix .

Merci à vous.

**Reply by dbarzin** [reply]
On peut déjà déclarer une adresse IP dans le formulaire des serveurs logiques. Il existe un lien entre les serveurs logiques au niveau des sous-réseaux. Peux-tu préciser ton idée ?

**Reply by ghost** [reply]
J'ai fais une capture d'écran et 2 petits exemples trés simple avec visio pour vous représenter mon idée.

Mercator:
![mercator_gw_example](https://user-images.githubusercontent.com/105782023/171819179-e1bb55b1-fc48-400d-95e7-acfb28472c88.jpg)

Visio exemple1:
![benj_mercator_exemple_01](https://user-images.githubusercontent.com/105782023/171818636-c33cf779-46e9-4698-b645-6875f37777d3.jpg)

Visio exemple2:
![benj_mercator_exemple_02](https://user-images.githubusercontent.com/105782023/171818657-ce28bf2a-70d2-4961-9f8e-bc15d3df160f.jpg)

Dans les exemples visios on voit les serveurs et les réseaux apparaissent bien derrière le firewall. 
Selon moi cela clarifie le schéma et la hierarchie entre les serveurs et leurs gateways et davantage visible.

Une fois de plus je peux vous contacter et vous faire une démo si cela peut simplifier la compréhension de cette demande.

**Reply by dbarzin** [reply]
C'est un schéma des flux réseau que tu souhaites mettre en place en utilisant l'information des interfaces firewall.
C'est tout à fait possible. Avec GraphViz on peut représenter dans des blocs les sous-réseau et les équipements qui en font partie mais un équipement ne peut pas appartenir à deux blocs distincts.

**Reply by ghost** [reply]
Si tu compare le schéma mercator et le schéma ci-dessous, la grosse différence c'est la hierarchisation des informations.
Le serveur appartient à un LAN qui se connecte à un firewall. 
L'avantage c'est que l'on voit tout de suite trés clairement l'architecture réseau.
D'ailleurs j'ai laissé une erreur dans mon schema qui doit porter à confusion. 

Voici un schema plus clair.
![benj_mercator_exemple_03](https://user-images.githubusercontent.com/105782023/173021118-27cdba1b-a14a-47e8-ab83-5bd6569c3c2d.jpg)

Dans un premier temps je ne pense pas qu'il soit nécessaire d'afficher les flux réseau. 
De mon point de vue ce qui serait important c'est juste de hierarchiser les éléments du réseau logique en dessous de la passerelle.
L'idée étant de retrouver la structure:
Le serveur appartient à un LAN qui se connecte à un firewall. 
Et du coup à la fin on devrait se retrouver avec les firewalls qui font les points de liaison entre les différentes infra/vlan.

Excellente journée.

**Reply by dbarzin** [reply]
La proposition est de remplacer le lien entre une passerelle et les sous-réseaux par une lien entre une passerelle et les réseaux.
Et de représenter ce lien dans la vue de l'infrastructure logique.

**Comment by ghost** [comment]
Bonjour.

J'espère que vous allez bien.
Je me disais à l'instant qu'une petite option pour faire apparaître les IPs sur les schémas logiques serait également extrêmement utile.

Je vous souhaite une agréable journée.

**Reply by dbarzin** [reply]
:sparkles: Il est maintenant possible de faire apparaître les IPs sur les schémas logiques !

**Reply by ghost** [reply]
Super ! 
Merci pour votre retour et votre réactivité.
Je vais mettre à jour l'application tout de suite pour tester l'option.
